app/ui/browser-modern/views/browser/tabbar/tab-contents.jsx

```diff
@@ -127,8 +128,14 @@ TabContents.propTypes = {
 function mapStateToProps(state, ownProps) {
   const page = PagesSelectors.getPageById(state, ownProps.pageId);
   const pageIsPinned = PagesSelectors.getPagePinned(state, ownProps.pageId);
+  let title = page.title || page.meta.title || page.location || 'Loading...';
```

Comment:
That would make it a little easier to write unit tests for this… including for pages with empty titles and URLs as titles.